spark/src/main/scala/org/apache/spark/sql/delta/actions/DeletionVectorDescriptor.scala

```diff
@@ -132,7 +133,9 @@ case class DeletionVectorDescriptor(
     storageType match {
       case UUID_DV_MARKER =>
         val absolutePath = this.absolutePath(tableLocation)
-        this.copy(storageType = PATH_DV_MARKER, pathOrInlineDv = absolutePath.toString)
+        this.copy(
+          storageType = PATH_DV_MARKER,
+          pathOrInlineDv = SparkPath.fromPath(absolutePath).urlEncoded)
```

Comment:
IIUC, we escape the path here instead of absolutePath because it's using the data in pathOrInlineDV. The tables with DVs already in use do not correctly encode the paths so we must work around this issue by doing the encoding on load, where we convert the Path to string. 

If this is the case, can we
* Make this very clear in a comment why it's encoded here.
* Add a test with an "existing" table that is storing DVs with incorrectly encoded paths and making sure that this solution works.

Reply:
Now that you mention it, shouldn't this decision also make it to the protocol? It sounds like otherwise we'd have unreadable tables

Reply:
The protocol states that absolute DVs paths behave the same as the `path` field in Add/RemoveFile actions. So we should go with option 1 then.
Anyway, there's no code that uses absolute paths, yet, so this change shouldn't break anything.